a10_octavia/controller/worker/tasks/health_monitor_tasks.py

```diff
@@ -30,22 +30,26 @@ class CreateAndAssociateHealthMonitor(task.Task):
     """Task to create a healthmonitor and associate it with provided pool."""
 
     @axapi_client_decorator
-    def execute(self, health_mon, vthunder):
+    def execute(self, listeners, health_mon, vthunder):
         method = None
         url = None
         expect_code = None
+
         if health_mon.type in a10constants.HTTP_TYPE:
             method = health_mon.http_method
             url = health_mon.url_path
             expect_code = health_mon.expected_codes
         args = utils.meta(health_mon, 'hm', {})
 
+        if listeners:
```

Comment:
This isn't a required check. The health monitor can't be created with out the listener being there